mod_reforged/hooks/entity/tactical/humans/desert_devil.nut

```diff
@@ -38,23 +38,36 @@
 		this.m.Skills.add(::new("scripts/skills/perks/perk_relentless"));
 	}
 
-	q.assignRandomEquipment = @(__original) function()
+	q.assignRandomEquipment = @() function()
 	{
-		__original();
-		local weapon = this.getMainhandItem();
-		if (weapon != null)
+		if (this.m.Items.hasEmptySlot(::Const.ItemSlot.Mainhand))
```

Comment:
I believe all the "empty slot" checks are unnecessary, as it's a freshly spawned unit and all the slots are in fact empty.